File: frontend/src/types/creator.ts
/**
 * 创作者主页库相关的类型定义。
 *
 * 字段命名与后端 API 保持一致（snake_case），不做 camelCase 转换。
 */

import type { CrawlPlatform } from './crawler'

/** 一位收藏的创作者 */
export interface Creator {
  id: number
  platform: CrawlPlatform
  platform_label: string
  name: string
  /** 主页链接或 ID：抓取时原样透传给 MC 的 --creator_id */
  homepage: string
  tags: string[]
  remark: string
  created_at: string
  updated_at: string
}

/** 新增 / 编辑创作者的请求体（tags 由后端清洗：≤10 个、每项 ≤20 字） */
export interface CreatorPayload {
  platform: CrawlPlatform
  name: string
  homepage: string
  tags?: string[]
  remark?: string
}

/** 列表接口返回（量少不分页，与后端 CreatorListData 对齐） */
export interface CreatorListData {
  total: number
  items: Creator[]
}

File: backend/app/models/creator.py
"""创作者主页 ORM 模型。

素材抓取「创作者主页」模式的素材库：按平台维护常抓的创作者。
homepage 字段就是抓取时透传给 MC --creator_id 的值 —— 主页链接或纯 ID
都行（各平台能吃什么见前端 PLATFORM_SPECS 的 creatorHelp，抓取失败的
错误会出现在任务日志尾部）。
"""

from datetime import datetime

from sqlalchemy import JSON, DateTime, Index, Integer, String, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base
from app.models.content import utcnow


class Creator(Base):
    """创作者主页表。"""

    __tablename__ = "creators"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True, comment="主键")
    platform: Mapped[str] = mapped_column(
        String(20), nullable=False, comment="平台标识（CrawlPlatform.ALL）"
    )
    name: Mapped[str] = mapped_column(
        String(50), nullable=False, comment="展示名（博主昵称或自起名）"
    )
    homepage: Mapped[str] = mapped_column(
        String(500), nullable=False, comment="主页链接或 ID（透传给 MC --creator_id）"
    )
    tags: Mapped[list] = mapped_column(
        JSON, nullable=False, default=list, comment="标签列表（自由输入，如母婴/美食）"
    )
    remark: Mapped[str] = mapped_column(String(200), nullable=False, default="", comment="备注")
    created_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, default=utcnow, comment="创建时间（UTC）"
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, default=utcnow, onupdate=utcnow, comment="更新时间（UTC）"
    )

    __table_args__ = (
        # 同一平台下主页唯一：主页就是抓取时的身份标识，重复录入没有意义
        UniqueConstraint("platform", "homepage", name="uq_creators_platform_homepage"),
        Index("ix_creators_platform", "platform"),
    )

    def __repr__(self) -> str:
        return f"<Creator id={self.id} platform={self.platform!r} name={self.name!r}>"
